src/imitation/policies/interactive.py

```diff
@@ -1,36 +1,58 @@
+"""Interactive policies that query the user for actions."""
+
 import abc
-from typing import Optional, List
+import collections
+import typing
 
 import gym
 import matplotlib.pyplot as plt
 import numpy as np
+from stable_baselines3.common import vec_env
 
 import imitation.policies.base as base_policies
 from imitation.util import util
 
 
 class DiscreteInteractivePolicy(base_policies.NonTrainablePolicy, abc.ABC):
+    """Abstract class for interactive policies with discrete actions.
+
+    For each query, the observation is rendered and then the action is provided
+    as a keyboard input.
+    """
+
     def __init__(
         self,
         observation_space: gym.Space,
         action_space: gym.Space,
-        action_names: List[str],
-        action_keys: List[str],
+        action_keys_names: collections.OrderedDict,
         clear_screen_on_query: bool = True,
     ):
+        """Builds DiscreteInteractivePolicy.
+
+        Args:
+            observation_space: Observation space.
+            action_space: Action space.
+            action_keys_names: `OrderedDict` containing pairs (key, name) for every
+                action, where key will be used in the console interface, and name
+                is a semantic action name.
+            clear_screen_on_query: If `True`, console will be cleared on every query.
+        """
         super().__init__(
             observation_space=observation_space,
             action_space=action_space,
         )
 
         assert isinstance(action_space, gym.spaces.Discrete)
-        assert len(action_names) == len(action_keys) == action_space.n
-        # Names and keys should be unique.
-        assert len(set(action_names)) == len(set(action_keys)) == action_space.n
+        assert (
+            len(action_keys_names)
+            == len(set(action_keys_names.values()))
+            == action_space.n
+        )
 
-        self.action_names = action_names
-        self.action_keys = action_keys
-        self.action_key_to_index = {k: i for i, k in enumerate(action_keys)}
+        self.action_keys_names = action_keys_names
+        self.action_key_to_index = {
+            k: i for i, k in enumerate(action_keys_names.keys())
```

Comment:
Iterating over a dict gives you the keys by default (you can leave as-is if you want to be explicit about it)
```suggestion
            k: i for i, k in enumerate(action_keys_names)
```

Reply:
Yeah, in this case I slightly prefer to keep it.